File: board/Port.py


from typing import Optional, TypeVar


from database.queries import directions
from Enum import Enum
from ResourceType import ResourceType


Port = TypeVar("Port")
Ports = list[Port]
Settlement = TypeVar("Settlement")
Settlements = list[Optional[Settlement]]


class Port:
	Settlements = Enum("Port::Settlements", **{type["label"]: type["id"]-1 for type in directions.get_side_corners()})
	Types = ResourceType

	def __init__(self, id: str, type: int):
		self.id: int = id
		self.type: int = type

		self.settlements: Settlements = [None for _ in range(self.Settlements.length)]


	def __eq__(self, right: Port) -> bool:
		return self.id == right.id

File: Source/webpage/Scripts.ts/Game/Board/Port.ts


import { Directions, ResourceType } from "../../Types.d.js";


import { Settlement } from "./index.js";


import { DIRECTIONS } from "../../Globals.js";


export class Port
{
	static readonly Settlements: Directions = DIRECTIONS["Side's Corners"];

	id: number;
	type: ResourceType;
	settlements: [Settlement|null, Settlement|null, Settlement|null, Settlement|null, Settlement|null, Settlement|null];


	constructor(id: number, type: ResourceType)
	{
		this.id = id;
		this.type = type;

		this.settlements = [null, null, null, null, null, null];
	}
}
